src/d3a/models/strategy/infinite_bus.py

```diff
@@ -0,0 +1,74 @@
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+import sys
+
+from d3a.models.strategy.commercial_producer import CommercialStrategy
+from d3a.models.strategy import BidEnabledStrategy
+from d3a.d3a_core.exceptions import MarketException
+from d3a.models.const import ConstSettings
+from d3a.models.read_user_profile import read_arbitrary_profile, InputProfileTypes
+
+INF_ENERGY = int(sys.maxsize)
+
+
+class InfiniteBusStrategy(CommercialStrategy, BidEnabledStrategy):
+    parameters = ('energy_sell_rate', 'energy_buy_rate')
+
+    def __init__(self, energy_sell_rate=None, energy_buy_rate=None):
+        super().__init__()
+        self.energy_per_slot_kWh = INF_ENERGY
+        energy_buy_rate = ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE \
+            if energy_buy_rate is None else energy_buy_rate
+        energy_sell_rate = ConstSettings.GeneralSettings.DEFAULT_MARKET_MAKER_RATE \
+            if energy_sell_rate is None else energy_sell_rate
+        self.energy_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY, energy_sell_rate)
+        self.energy_buy_rate = read_arbitrary_profile(InputProfileTypes.IDENTITY, energy_buy_rate)
+
+    def offer_energy(self, market):
+        energy_rate = self.energy_rate[market.time_slot]
+        offer = market.offer(
+            self.energy_per_slot_kWh * energy_rate,
+            self.energy_per_slot_kWh,
+            self.owner.name
+        )
+        self.offers.post(offer, market)
+
+    def buy_energy(self, market):
+        for offer in market.sorted_offers:
+            if offer.seller == self.owner.name:
+                # Don't buy our own offer
+                continue
+            if (offer.price / offer.energy) <= self.energy_buy_rate[market.time_slot]:
+                try:
+                    self.accept_offer(market, offer)
+                except MarketException:
+                    # Offer already gone etc., try next one.
+                    continue
+
+    def event_tick(self, *, area):
+        super().event_tick(area=area)
+
+        if ConstSettings.IAASettings.MARKET_TYPE == 1:
+            for market in self.area.all_markets:
+                self.buy_energy(market)
+
+    def event_market_cycle(self):
+        super().event_market_cycle()
+        if ConstSettings.IAASettings.MARKET_TYPE == 2:
+            for market in self.area.all_markets:
+                self.post_bid(market, self.energy_rate[market.time_slot] * INF_ENERGY, INF_ENERGY)
```

Comment:
InfiniteBusStrategy should also be posting bids for PayAsClearMarket

Reply:
You are right! I was confused by this comment: https://gridsingularity.atlassian.net/browse/D3ASIM-1104?focusedCommentId=13855&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-13855 and only allowed the PAB. Will change this. Shall I add a separate integrationtest for that?

Reply:
Or how about via Scenario Outline to test the same integration for 2-sided PAB & PAC?

Reply:
@spyrostz @faizan2590 But how can I set the MarketType outside of the config file? You mean by setting it in the ConstSettings?

Reply:
Yes, you can run the same test 3 times by changing the ConstSettings.IAASettings.MARKET_TYPE parameter to run for different market types.

Reply:
@spyrostz but how can i change it in the setup file in an integration test?

Reply:
you can import the ConstSettings, and change the value in a step implementation, after you have read the setup file (this is an example, there are other ways to do that, ask if it is not to your liking). That way the value that you have defined in the setup file will be 'overwritten', since the ConstSettings are global for the simulation.